cli/factory.rs

```diff
@@ -137,7 +137,7 @@ struct CliFactoryServices {
   maybe_import_map: Deferred<Option<Arc<ImportMap>>>,
   maybe_inspector_server: Deferred<Option<Arc<InspectorServer>>>,
   root_cert_store_provider: Deferred<Arc<dyn RootCertStoreProvider>>,
-  blob_store: Deferred<BlobStore>,
+  blob_store: Deferred<Arc<BlobStore>>,
```

Comment:
I moved the reference counter outside of `BlobStore` rather than on each of its fields because it makes more sense and resembles the other service fields. Ended up changing a lot of lines though.